src/targets/gpu/fuse_mlir.cpp

```diff
@@ -423,6 +470,97 @@ std::vector<instruction_ref> mlir_contiguous(module_pass_manager& mpm,
     return result;
 }
 
+struct find_mlir_split_reduce
+{
+    mlir_mode conv_mode = mlir_mode::none;
+    mlir_mode dot_mode  = mlir_mode::none;
+    auto matcher() const
+    {
+        auto dot_or_conv = match::name("gpu::mlir_op");
+        // TODO: Handle reshapes inbetween
+        return mlir_split_reduce()(match::any_of[match::inputs()](dot_or_conv.bind("gemm")));
+    }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto reduce_ins = r.result;
+        auto gemm_ins   = r.instructions["gemm"];
+        auto* rm        = reduce_ins->module_inputs().front();
+        auto names      = rm->get_parameter_names();
+        std::sort(names.begin(), names.end());
+        module_ref gemm_old_mm = gemm_ins->module_inputs().front();
+        module_ref mm          = mpm.create_module(gemm_old_mm->name() + "_split_fused_reduce");
```

Comment:
You can do `mpm.create_module(gemm_old_mm->name() + "_split_fused_reduce", gemm_ins->module_inputs().front())` to copy the module with a new name.

Reply:
Done.